### ISSUE TITLE
I cannot uninstall kernels

### ISSUE THREAD

**Issue by hiredman** [systemd/systemd#28448]
### systemd version the issue has been seen with

254

### Used distribution

Fedora Rawhide

### Linux kernel version used

6.4.0-0.rc4.20230601git929ed21dfdb6.38.fc39.x86_64

### CPU architectures issue was seen on

x86_64

### Component

other

### Expected behaviour you didn't see

I expected running ` dnf remove kernel-core-6.5.0-0.rc1.11.fc39.x86_64 ` to remove an old kernel I have installed

### Unexpected behaviour you saw

the dnf remove operation fails with a not super helpful error message, but /var/log/dnf5.log ends up containing this:
```
2023-07-18T21:09:10+0000 [2477001] INFO [scriptlet] + /bin/kernel-install remove 6.5.0-0.rc1.11.fc39.x86_64 /lib/modules/6.5.0-0.rc1.11.fc39.x86_64/vmlinuz
2023-07-18T21:09:10+0000 [2477001] INFO [scriptlet] Too many arguments.
2023-07-18T21:09:10+0000 [2477001] INFO [scriptlet] + exit 1
```
so it appears kernel-install is calling something else, and when it does it is passing too many arguments. way out of my depth here

### Steps to reproduce the problem

I have no idea what kernel-install is actually doing, so hard to say what state I might have that is required to reproduce this, but this reproduces for me trying to uninstall 3 out of the 4 kernels I have installed (using the 4th one, don't want to uninstall it).

**Comment by bluca** [comment]
can you run that same kernel-install command after exporting SYSTEMD_LOG_LEVEL=debug and attach the full output?

**Comment by hiredman** [comment]
perfect, I figured there must be some debugging knob I didn't know of, the additional output looks like:
```
2023-07-18T22:44:18+0000 [2521331] INFO [scriptlet] + /bin/kernel-install remove 6.5.0-0.rc2.17.fc39.x86_64 /lib/modules/6.5.0-0.rc2.17.fc39.x86_64/vmlinuz
2023-07-18T22:44:18+0000 [2521331] INFO [scriptlet] Loading /usr/lib/kernel/install.conf…
2023-07-18T22:44:18+0000 [2521331] INFO [scriptlet] Loaded /usr/lib/kernel/install.conf.
2023-07-18T22:44:18+0000 [2521331] INFO [scriptlet] Loading /etc/machine-info…
2023-07-18T22:44:18+0000 [2521331] INFO [scriptlet] MACHINE_ID=0f4bb1a6af98430b8f2c31c97928c6c3 set via /etc/machine-info.
2023-07-18T22:44:18+0000 [2521331] INFO [scriptlet] Found cgroup2 on /sys/fs/cgroup/, full unified hierarchy
2023-07-18T22:44:18+0000 [2521331] INFO [scriptlet] Found container virtualization none.
2023-07-18T22:44:18+0000 [2521331] INFO [scriptlet] /dev/nvme0n1p2: Partitition has wrong PART_ENTRY_TYPE=0fc63daf-8483-4772-8e79-3d69d8477de4 for XBOOTLDR partition.
2023-07-18T22:44:18+0000 [2521331] INFO [scriptlet] Couldn't find an XBOOTLDR partition.
2023-07-18T22:44:18+0000 [2521331] INFO [scriptlet] Failed to check file system type of "/efi": No such file or directory
2023-07-18T22:44:18+0000 [2521331] INFO [scriptlet] File system "/boot" is not a FAT EFI System Partition (ESP) file system.
2023-07-18T22:44:18+0000 [2521331] INFO [scriptlet] Using EFI System Partition at /boot/efi as $BOOT_ROOT.
2023-07-18T22:44:18+0000 [2521331] INFO [scriptlet] Using entry token: 0f4bb1a6af98430b8f2c31c97928c6c3
2023-07-18T22:44:18+0000 [2521331] INFO [scriptlet] Too many arguments.
2023-07-18T22:44:18+0000 [2521331] INFO [scriptlet] + exit 1
```

**Comment by bluca** [comment]
@yuwata @keszybz any idea?

**Comment by damaestro** [comment]
I've ran into this also. The previous `kernel-install` (bash) silently ignored the extra argument that the C based implementation is complaining about. I've opened a bug for the kernel-core package to fix the spec, but there will need to be backwards compatibility to deal with all of the old scriptlets out there.

https://bugzilla.redhat.com/show_bug.cgi?id=2223794

**Comment by damaestro** [comment]
@hiredman (and anyone else that runs into this),

You can remove the affected kernel-core package and manually run the kernel-install remove command successfully.

```
sudo rpm -e --noscripts kernel-core-6.5.0-0.rc1.11.fc39.x86_64
sudo kernel-install remove 6.5.0-0.rc1.11.fc39.x86_64
```

**Comment by yuwata** [comment]
Thank you for the report. Fix is waiting in #28450.

**Comment by hiredman** [comment]
fantastic, thanks everyone
